### DISCUSSION TITLE
Better quiz sliver backend

### DISCUSSION THREAD

**Discussion by nmanumr** [nmanumr/quizsliver#13] [Ideas]
For now, I'm using firebase as the backend server which was always a hurdle in the development of some new feature. We can write a custom backend server with some proper database which will allow us to make a lot of improvements and unleash a lot of new possibilities.

Here are a few ideas that we can have with the new backend server:

1. We can have a stand-alone open API that can be used by anyone to search quiz MCQ answers. This way we don't have to rely on this version of quiz sliver extension only but anyone can write their own version of quiz sliver. Imagine a quiz sliver mobile app.
2. For now we search for questions totally on the basis of their hash, with our own backend we can use some better text-based search algorithms or maybe some AI-based solution too.
3. For now, we only store one correct answer (which can be wrong, infact, an attack is possible which will corrupt all the correct answers) but with this new backend, we can say what the chances of this option being correct.
4. We can export MCQ questions to any format like JSON, HTML, or PDF.

Ok, I'm out of ideas here but the possibilities are endless. Please let me know what you think and if I get some good response I will start working on this new backend and maybe fix a lot of known issues with the extension too.

PS: Writing at 3 AM with my half-dead brain so expect some mistakes.

**Comment by nmanumr** [comment]
BTW I had a dream to write a full suite of tools specific to COMSATS students specifically which would include:
* A timetable app (I worked really hard on this [here is the source](https://github.com/BlackArcX/timetable-backend)) the idea was a mobile app that can show your timetable right on your home screen and can also help you arrange makeup classes.
* A fun app to rate and review teachers so students don't have to spam the help desk at the start of every semester.
* CUOnlien utils which include:
   - quiz-sliver
   - auto feedback filler ([source](https://gist.github.com/nmanumr/9037d9b3304f74d5515ff0af691ed850))
   - AI-based GPA forecast
* An awesome list of things for COMSATS students (and [here it is](https://github.com/nmanumr/awesome-comsats))

As you can see I'm done with most of the things here and the rest of the stuff is also not that difficult to do. All I need is a few people who can spread the word around and support the development, and I'm not asking for financial support either just some moral support, a few words of appreciation publicly or personally, or just showing some interest would do the magic.

**Reply by ossamamehmood** [reply]
Yup, You're aiming 💯. There should be an interface for quizsliver extension as well. Right know there's no interface wizard while clicking the extension.

**Comment by Talha-glitched** [comment]
Can we make this quiz compatible with Abbottabad Campus in any way?

**Reply by nmanumr** [reply]
that would require some effort on the parser level rest of the code can work as it is. If you are willing to contribute I can guide you.

**Comment by Talha-glitched** [comment]
I would love to be guided but I am in 3rd semester and currently learning
java, I know C language but if you can guide me I can learn anything

**Reply by ossamamehmood** [reply]
It requires knowledge of Javascript in order to understand the code first then towards you could build that!

**Reply by ossamamehmood** [reply]
However, a better idea to acknowledge your administrator to have a **CUONLINE** PORTAL like other campuses would be greater!

**Comment by Talha-glitched** [comment]
So I can’t do anything :( I’ll come back when I’ve learnt js. if you have
the time to do it for me it’s a long shot but it’s worth a try for me

**Reply by nmanumr** [reply]
Definitely man I willing to guide you just start learning javascript from anywhere you like (I personally like https://javascript.info/). Also, you need to learn about the basics of web scraping to understand this project codebase.

And, the other thing you can do is to save the snapshot of cuonline pages like MCQs page, answers/score page etc. By snapshot I mean save the page with CTRL+S. These snapshots will letter help us to write parser on letter stages.
